.github/workflows/cluster-integration-tests-kind.yaml

```diff
@@ -0,0 +1,41 @@
+name: Cluster-based Integration Tests (KIND)
+on: [push, pull_request]
+jobs:
+  integration-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          token: ${{ github.token }}
+      - name: Install Helm
+        uses: azure/setup-helm@v1
+      - name: Install Argo
+        run: |
+          # Download the binary
+          curl -sLO https://github.com/argoproj/argo-workflows/releases/download/v3.1.1/argo-linux-amd64.gz
+          # Unzip
+          gunzip argo-linux-amd64.gz
+          # Make binary executable
+          chmod +x argo-linux-amd64
+          ./argo-linux-amd64 version
+      - name: Create Kind Cluster
+        uses: helm/kind-action@v1.1.0
+        with: 
+          version: v0.11.1
+          node_image: kindest/node:v1.21.1
+          cluster_name: adelphi-integration-tests
+          config: helm/adelphi/tests/kind-config-3-workers.yaml
+      - name: Install Chart and Run Workflow
+        run: helm install adelphi helm/adelphi -n cass-operator -f helm/adelphi/tests/kind-values-trunk.yaml --debug --wait || true
+      - name: Install Chart and Run Workflow (2nd attempt in case of timeout)
```

Comment:
In a constrained environment, it might take over a minute to install the two CRDs we currently have. If it times out, it won't complete the Adelphi installation and we have to try again, so this is what this step does.
On the other hand, it the first installation succeeded, it will simply ignore the second attempt.

In future If we can contribute a patch to helm such that the timeout is configurable, we can remove this step.

Reply:
Here you're talking a timeout simply while installing the chart?  Right?  I think that's the case, just want to confirm.  

The timeout they do have on the cli tool won't cover the case you're talking about here? With the initial installation of the CRDs?

```
--timeout duration             time to wait for any individual Kubernetes operation (like Jobs for hooks) (default 5m0s)
```

Reply:
@jdonenine I confirmed my suspicion that the `--timeout` flag only applies to the regular resources, not to the CRDs.
Installing the chart with:

```
helm install adelphi helm/adelphi -n cass-operator -f helm/adelphi/tests/kind-values-trunk.yaml --debug --wait --timeout=10m
```

Prints the following for the CRDs section:

```
wait.go:47: [debug] beginning wait for 29 resources with timeout of 1m0s
```

And the following for the other resources:

```
wait.go:47: [debug] beginning wait for 26 resources with timeout of 10m0s
```

It seems the timeout for the CRDs is static at 1min. We can always revisit this in the future when it becomes tunable.

Reply:
Well that's unfortunate 😔 

Think we should file an enhancement request with helm to allow for the additional timeouts to be set related to CRD install?